distribution/pull_v2.go

```diff
@@ -1027,3 +1028,11 @@ func toOCIPlatform(p manifestlist.PlatformSpec) specs.Platform {
 		OSVersion:    p.OSVersion,
 	}
 }
+
```

Comment:
should this be included in upstream's default spec? if so, we should consider un-exporting this function, or adding a note in the comment that it's temporary / deprecated

Reply:
It should not replace `DefaultSpec` as they have different use cases. I can try to see what they think about an additional method. In https://github.com/containerd/containerd/pull/6455#issuecomment-1016696041 seemed like they wanted it, although containerd itself does not really have a use-case for this function atm.

Reply:
I guess alternatively, github.com/tonistiigi/go-archvariant could have a helper for this, that way we could still use the same function in both places, without introducing something new in `distribution`, especially if it's (possibly) temporary (I'm trying to trim down the exported "api" of the distribution package)

Would that work for you?

Reply:
Ideally, I'd like to not add dependencies to `go-archvariant` or expand its scope.

Reply:
as it's (hopefully) temporary; perhaps we should just copy it to both places and make it private